contracts/BatchExchangeViewer.sol

```diff
@@ -144,6 +145,41 @@ contract BatchExchangeViewer {
         return (elements, nextPageUser, nextPageUserOffset);
     }
 
+    function getEncodedOrdersPaginated(address previousPageUser, uint16 previousPageUserOffset, uint256 pageSize)
+        public
+        view
+        returns (bytes memory)
+    {
+        bytes memory elements = new bytes(pageSize * AUCTION_ELEMENT_WIDTH);
+        uint16 currentOffset = previousPageUserOffset;
+        uint256 index = 0;
+        address currentUser = previousPageUser;
+        while (index < pageSize) {
+            bytes memory element = batchExchange.getEncodedUserOrdersPaginated(currentUser, currentOffset, 1);
+            if (element.length > 0) {
+                currentOffset += 1;
+                for (uint256 i = 0; i < element.length; i++) {
+                    elements[(index * AUCTION_ELEMENT_WIDTH) + i] = element[i];
+                }
+                index += 1;
+            } else {
+                currentOffset = 0;
+                bytes memory nextUser = batchExchange.getUsersPaginated(currentUser, 1);
+                if (nextUser.length > 0) {
+                    currentUser = nextUser.toAddress(0);
+                } else {
+                    break;
+                }
+            }
+        }
+        //setLength(elements, index * AUCTION_ELEMENT_WIDTH);
```

Comment:
Should this be removed?

Reply:
It should be uncommented and the assembly right below should be removed.

Reply:
Even better.